src/pages/product-details/product-details.tsx

```diff
@@ -61,17 +63,31 @@ export default function ProductDetails() {
       {!product ? (
         <h1>The product could not be found</h1>
       ) : (
-        <div>
+        <div className="">
           <div className="header">
             <div className="heading">
               <h1>Product: {product.name}</h1>
             </div>
             <div className="buttons">
-              <button onClick={() => handleAddToCart(product)}>
+              <button
+                onClick={() => handleAddToCart(product)}
+                disabled={auth.customer?.role !== "customer"}
```

Comment:
Create a property in the context to get the role and use it here like: isSimpleUser or isAdmin